pkg/scheduler/factory.go

```diff
@@ -343,10 +343,13 @@ func MakeDefaultErrorFunc(client clientset.Interface, podLister corelisters.PodL
 			klog.Warningf("Pod %v/%v doesn't exist in informer cache: %v", pod.Namespace, pod.Name, err)
 			return
 		}
-		// As <cachedPod> is from SharedInformer, we need to do a DeepCopy() here.
-		podInfo.Pod = cachedPod.DeepCopy()
-		if err := podQueue.AddUnschedulableIfNotPresent(podInfo, podQueue.SchedulingCycle()); err != nil {
-			klog.Error(err)
+
+		if len(cachedPod.Spec.NodeName) == 0 {
```

Comment:
I am doubt that this change will address your issue, imagine that pod's nodename is set after this step,  the pod will be  added back to the Q again.

Reply:
I think this may happen, but the next time this pod is scheduled, it will be corrected.